types/swagger-parser/index.d.ts

```diff
@@ -1,7 +1,50 @@
 // Type definitions for swagger-parser 4.x
 // Project: https://www.npmjs.com/package/swagger-parser
 // Definitions by: Tobias Wolff <https://github.com/Tobias4872/>
+//                 Eric Henry <https://github.com/EricHenry/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export function parse(api: any, options?: any): Promise<any>;
-export function parse(api: any, options: any, callback: (err: any, result?: any) => void): void;
+export namespace SwaggerParser {}
+
+export interface Options {
+    allow?: {
+        json?: boolean;
+        yaml?: boolean;
+        empty?: boolean;
+        unknown?: boolean;
+    };
+    $ref?: {
+        internal?: boolean;
+        external?: boolean;
+        circular?: boolean | 'ignore';
+    };
+    validate?: {
+        schema?: boolean;
+        spec?: boolean;
+    };
+    cache?: {
+        fs?: number;
+        http?: number;
+        https?: number;
+    };
+}
+
+export function parse(api: string | any, options?: Options): Promise<any>;
```

Comment:
I think this is a good idea. There is an [official swagger schema](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/swagger-schema-official) type defined in the DefinitelyTyped project. Is there a supported way to reference that type, inside of this project definition?